src/providers/cortex/trulens/providers/cortex/endpoint.py

```diff
@@ -14,9 +15,51 @@
 pp = pprint.PrettyPrinter()
 
 
+# the credit consumption table needs to be kept up-to-date with
+# the latest cost information https://www.snowflake.com/legal-files/CreditConsumptionTable.pdf#page=9.
+CORTEX_MODEL_TO_CREDITS_PER_1M_TOKENS = {
+    "reka-core": 5.5,
+    "mistral-large": 5.1,
+    "llama3.1-405b": 5,
+    "llama3-70b": 1.21,
+    "llama3.1-70b": 1.21,
+    "snowflake-arctic": 0.84,
+    "jamba-instruct": 0.83,
+    "llama2-chat-70b": 0.45,
+    "reka-flash": 0.45,
+    "mixtral-8x7b": 0.22,
+    "llama3-8b": 0.19,
+    "llama3.1-8b": 0.19,
+    "mistral-7b": 0.12,
+    "gemma-7b": 0.12,
+}
+
+
 class CortexCallback(EndpointCallback):
     model_config: ClassVar[dict] = dict(arbitrary_types_allowed=True)
 
+    # TODO (Daniel): cost tracking for Cortex finetuned models is not yet implemented.
+
+    def _compute_credits_consumed(
+        self, cortex_model_name: str, n_tokens: int
+    ) -> float:
+        try:
+            if cortex_model_name in CORTEX_MODEL_TO_CREDITS_PER_1M_TOKENS:
+                return (
+                    CORTEX_MODEL_TO_CREDITS_PER_1M_TOKENS[cortex_model_name]
+                    * n_tokens
+                    / 1e6
+                )  # we maintain config per-1M-token cost
+            else:
+                raise ValueError(
+                    f"Model {cortex_model_name} not valid or not supported yet."
```

Comment:
can you say "not supported yet for cost estimation" or something? I'm mainly worried that some new model will come out and it'll work to run it, but then this message will make it sound like it can't even run.

Also, now that I'm thinking about it, when a new model comes out, does this mean until the next release ppl can't use the model because this will fire? Maybe we can find a place where `CORTEX_MODEL_TO_CREDITS_PER_1M_TOKENS` is kept up to date, or at least is not tied to our release?

Reply:
I have been trying to find a more up-to-date / real time place than https://www.snowflake.com/legal-files/CreditConsumptionTable.pdf#page=9 but no luck so far :/

Reply:
I'm moving to using a static file instead of in code per our discussion